NightscoutServiceKit/OTPManager.swift

```diff
@@ -202,27 +237,44 @@ public class OTPManager {
     }
     
     enum OTPValidationError: LocalizedError {
-        case expired
-        case previouslyUsed
+        case missingOTP
+        case expired(deliveryDate: Date, maxOTPsToAccept: Int)
+        case previouslyUsed(otp: String)
+        case incorrectOTP(otp: String)
         case invalidFormat(otp: String)
         case codeGeneratorFailed
         
         var errorDescription: String? {
             switch self {
-            case .expired:
-                return "OTP code is expired."
-            case .previouslyUsed:
-                return "OTP code was previously used"
+            case .missingOTP:
+                return "Error: OTP is required."
+            case .expired(let deliveryDate, let maxOTPsToAccept):
+                let dateFormatter = DateFormatter()
+                dateFormatter.dateFormat = "h:mm"
+                return String(format: "Error: OTP sent at %@ has expired. Only the last %u OTPs are accepted. See LoopDocs for troubleshooting.", dateFormatter.string(from: deliveryDate), maxOTPsToAccept)
+            case .previouslyUsed(let otp):
+                return "Error: OTP \(otp) was already used. Wait for a new OTP for each command."
             case .invalidFormat(let otp):
-                return "OTP code has an invalid format: \(otp)"
+                return "Error: OTP has an invalid format: \(otp)."
+            case .incorrectOTP(let otp):
+                return "Error: OTP is incorrect: \(otp)."
             case .codeGeneratorFailed:
-                return "OTP codes could not be generated."
+                return "Error: OTP validation is not available. See LoopDocs to setup."
             }
         }
     }
     
 }
 
```

Comment:
Several of these methods in this class, and tests, use these structures to wrap up OTPs and their associated  valid periods.